idunn/geocoder/models/geocodejson.py

```diff
@@ -149,22 +151,46 @@ class Feature(BaseModel):
     properties: FeatureProperties
 
     # From CanalTP/mimirsbrunn:libs/bragi/src/model.rs
-    distance: Optional[PositiveInt]
+    distance: Optional[int]
     context: Optional[Context]
 
 
+class IntentionFilter(BaseModel):
+    q: Optional[str]
+    bbox: Optional[Rect]
+    category: Optional[str]
+    source: Optional[str]
```

Comment:
No, when absent the source can still be determined by the /places/ endpoint (depending on the location, etc.)  
This field is defined to allow the intention detection to force the source to be used instead of a default one that would be selected by the /places/ endpoint.